extensions/film/schema.json

```diff
@@ -0,0 +1,69 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://linz.github.io/stac/_STAC_VERSION_/film/schema.json",
+  "title": "Film Extension",
+  "description": "STAC Film Extension for STAC Items.",
+  "allOf": [
+    {
+      "type": "object",
+      "required": ["type", "properties", "assets"],
+      "properties": {
+        "type": {
+          "const": "Feature"
+        },
+        "properties": {
+          "required": ["film:id", "film:sequence_number"],
+          "$ref": "#/definitions/fields"
+        },
+        "assets": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/fields"
+          }
+        }
+      }
+    },
+    {
+      "$ref": "#/definitions/stac_extensions"
+    }
+  ],
+  "definitions": {
+    "stac_extensions": {
```

Comment:
This shouldn't be necessary, as done in https://github.com/linz/stac/pull/58. Basically, if we're validating against this schema then the file contains a reference to this extension, so it's tautological to validate that we're referencing the extension.

Reply:
Nice, I thought that looked weird.

Reply:
I disagree with this, I also believe #58 should not have been merged. We need to have wider review of sweeping changes before merging things..

By not having the schema referenced in the extensions  you could validate a random stac file against this schema and it would be considered "valid" even though it is invalid as stac_extension was not referenced.

All of the STAC spec extensions I looked at have this reference, I do not think we should be deviating from their standard.

eg https://github.com/stac-extensions/raster/blob/main/json-schema/schema.json#L72

Reply:
Why would you validate a file against a schema which is not referenced within that file?

Reply:
Asked [upstream](https://github.com/stac-extensions/template/issues/10).

Reply:
Because that's how most JSON schemas work? JSON documents generally never reference the schema they are being validated against.

I would also try and keep as close to the STAC examples as possible.

Reply:
> Because that's how most JSON schemas work? JSON documents generally never reference the schema they are being validated against.

STAC has an explicit mechanism for referencing the schemas though. I would've thought not referencing the schema would mean the file isn't a STAC document.

Reply:
I don't really want this issue to hold up this PR. Sounds like https://github.com/linz/stac/pull/58 should be reversed pending reasoning/advice from STAC Spec maintainers, with discussion reopened on a new PR to remove if necessary?

Reply:
Why not leave it as is? Nothing's broken, and it's not possible to write a negative test for that part of the old schema (which is how this got started).

Reply:
I think the default thing we should do when concern is raised over a merged PR (especially one that is yet to be released) is to revert and discuss.

Reply:
The [current example](https://github.com/linz/stac/blob/41a13241e28e3521cf072a947398eabacc590415/extensions/linz/examples/collection.json) of a valid `collection.json` on the `linz` extension doesn't reference the `quality` extension in `stac_extensions`. Should it? Would that have been picked up if the `schema.json` for the `linz` extension contained this check?

Reply:
It doesn't because it doesn't need to, and therefore shouldn't, as explained [upstream](https://github.com/stac-extensions/template/issues/10). Having to list nested extensions is pointless overhead.

Reply:
Why does it list the `version` extension?

Reply:
Because, like the template repo, the version extension requires us to reference it in every STAC file even though it's referenced from the LINZ extension.